### PR TITLE
Lint some tools

### PR DESCRIPTION
**Pull request by nsoranzo**
Lint a number of tools with enhancements like:
- drop deprecated `interpreter` attribute of `<command>`
- use `format_source` instead of `format` in `<data>`
- single-quote variables referring to paths and text params in `<command>`
- add `<citation>`
- fix various syntax errors.

Also some fixes and enhancements for `galaxy.xsd` found while linting the tools.

### CODE REVIEW

**Comment by jmchilton** [comment]
Looks good to me -thanks a bunch!

File: tools/evolution/add_scores.xml [hunk 1]
```diff
@@ -13,13 +15,8 @@
   </inputs>
 
   <outputs>
-    <data format="input" name="out_file1" />
+    <data format_source="input1" name="out_file1" />
   </outputs>
-
-  <requirements>
-    <requirement type="package">add_scores</requirement>
```
Comment (bgruening) [inline tools/evolution/add_scores.xml:20]: Can/Should we depend on python here?

File: tools/filters/pasteWrapper.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <tool id="Paste1" name="Paste" version="1.0.0">
   <description>two files side by side</description>
-  <command interpreter="perl">pasteWrapper.pl $input1 $input2 $delimiter $out_file1</command>
+  <command>perl '$__tool_directory__/pasteWrapper.pl' '$input1' '$input2' $delimiter '$out_file1'</command>
```
Comment (bgruening) [inline tools/filters/pasteWrapper.xml:3]: perl dependency?

File: tools/metag_tools/shrimp_color_wrapper.xml [hunk 3]
```diff
@@ -1,12 +1,15 @@
 <tool id="shrimp_color_wrapper" name="SHRiMP for Color-space" version="1.0.0">
-  <description>reads mapping against reference sequence </description>
-  <command interpreter="python">
-    #if $param.skip_or_full=="skip" #shrimp_color_wrapper.py $input_target $input_query $output1 
-    #else                           #shrimp_color_wrapper.py $input_target $input_query $output1 $param.spaced_seed $param.seed_matches_per_window $param.seed_hit_taboo_length $param.seed_generation_taboo_length $param.seed_window_length $param.max_hits_per_read $param.max_read_length $param.kmer $param.sw_match_value $param.sw_mismatch_value $param.sw_gap_open_ref $param.sw_gap_open_query $param.sw_gap_ext_ref $param.sw_gap_ext_query $param.sw_crossover_penalty $param.sw_full_hit_threshold $param.sw_vector_hit_threshold  
-    #end if#
-  </command>
+    <description>reads mapping against reference sequence </description>
+    <requirements>
+      <requirement type="binary">rmapper-cs</requirement>
```
Comment (bgruening) [inline tools/metag_tools/shrimp_color_wrapper.xml:4]: a potential candidate for the dep-mapping file

**Review by bgruening — APPROVED** [review]
would be nice to fix the dependencies as well, but this can also happen in a different PR

**Comment by nsoranzo** [comment]
> would be nice to fix the dependencies as well, but this can also happen in a different PR

@bgruening I'd prefer to have the requirement fixes in a separate PR, where we can fix them all at once using the same Perl/Python/whatever version across all these tools. Or when we do the final tool migration...

**Comment by nsoranzo** [comment]
I've restarted the failed tests, it's all green now.

**Comment by bgruening** [comment]
Thanks @nsoranzo!

**Pull request merged**